test_util/aws.py

```diff
@@ -99,6 +97,18 @@ def create_stack(self, name, template_url, parameters, deploy_timeout=60):
             # so that we may pass parameters directly from yaml (which parses numbers as non-strings)
             Parameters=param_dict_to_aws_format(parameters))
 
+    def create_stack_from_string(self, name, template, parameters, deploy_timeout=60):
```

Comment:
Nope, cannot default to empty string either, because if TemplateBody is given (even as an empty string) it will be taken over TemplateURL:
```
Invalid length for parameter TemplateBody, value: 0, valid range: 1-inf
```

boto3 can just be viewed as a wrapper for the XML HTTP API. Or rather, the documentation, and the library itself are just wrappers around schemas, so all paths will likely just lead back here:
https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_CreateStack.html